### PR TITLE
core: Update dirty BitmapData before rendering

### PR DESCRIPTION
**Pull request by Aaron1011**
We now have a `MutationContext` available, so this doesn't need to be done in `run_frame`

### CODE REVIEW

**Comment by SN902** [comment]
good news: this commit brings improvements to rainbow and wind effects
bad news: black background 
[Rd_flying.zip](https://github.com/ruffle-rs/ruffle/files/9538890/Rd_flying.zip)

File: core/src/display_object/bitmap.rs [hunk 1]
```diff
@@ -248,32 +248,30 @@ impl<'gc> TDisplayObject<'gc> for Bitmap<'gc> {
...
+                    log::error!("Failed to update dirty bitmap {:?}: {:?}", bitmap_handle, e);
+                }
                 drop(bd);
-                bitmap_data.write(context.gc_context).set_dirty(false);
+                inner_bitmap_data.write(context.gc_context).set_dirty(false);
             }
-        }
-    }
 
-    fn render_self(&self, context: &mut RenderContext) {
```
Comment (adrian17) [inline core/src/display_object/bitmap.rs:269]: out of curiosity: how come `RenderContext` here didn't need template args before?

**Review by adrian17 — APPROVED** [review]

**Comment by adrian17** [comment]
I'm assuming this also fixes blood flickering in Binding of Isaac - great :D

File: core/src/display_object/bitmap.rs (hunk 1 above)
Comment (Aaron1011) [reply core/src/display_object/bitmap.rs:269]: We weren't using `gc_context`, so the `'gc` lifetime didn't have to match anything.

**Comment by Aaron1011** [comment]
I can reproduce the black background - issue - I'm investigating.

**Comment by Aaron1011** [comment]
The 'Rd_flying' bug is actually a separate issue (unimplemented `BlendMode.SCREEN`) which happens to be exposed by this PR.

**Review by Herschel — APPROVED** [review]
Thank you!

**Pull request merged**